channeldb/db.go

```diff
@@ -9,16 +9,41 @@ import (
 	"sync"
 
 	"github.com/boltdb/bolt"
+	"github.com/go-errors/errors"
 	"github.com/roasbeef/btcd/btcec"
 	"github.com/roasbeef/btcd/chaincfg"
 	"github.com/roasbeef/btcd/wire"
 )
 
 const (
-	dbName = "channel.db"
+	dbName           = "channel.db"
+	dbFilePermission = 0600
+
+	// dbCorruptedPrefix is used for renaming the default name of the
+	// database if migration is failed.
+	dbCorruptedPrefix = "corrupted."
+
+	// dbBackupPrefix is used for constructing the name of the backup
+	// file of the database during migration process.
+	dbBackupPrefix = "backup."
 )
 
+// Migration is a function which changes the data accordingly to maintain the
+// proper state of the database.
+type Migration func(d *DB) error
+
+type Version struct {
+	number    uint32
+	migration Migration
+}
+
 var (
+	// DBVersions is storing all versions of database. If current version of
+	// database don't match with latest version this list will be used for
+	// retrieving all migration function that are need to apply to the
+	// current db.
+	DBVersions = []Version{{1, nil}}
```

Comment:
Like the others above, I don't think this variable needs to be publicly exported.